index.js

```diff
@@ -168,6 +168,18 @@ function session(options){
     // req.secret is passed from the cookie parser middleware
     var secrets = secret || [req.secret];
 
+    // Since we execute the functions (if they are a secet) we need
+    // to make a copy so that we don't overwrite the original function
+    var processedSecrets = [];
+
+    for (var i = 0; i < secrets.length; i++) {
```

Comment:
The error  "secret option array must contain one or more strings" is now misleading. Would you be willing to make the following changes:

1. Do not accept `req.secret` to be a function.
2. Only accept a single `function` for the secret instead of letting any element in the array be a function? The function can return an array if it wants to provide multiple.